java/src/main/java/org/micromanager/lightsheet/StackResampler.java

```diff
@@ -574,13 +594,33 @@ public TaggedImage next() {
       return () -> StreamSupport.stream(Spliterators.spliterator(iterator,
                         StackResampler.this.cameraImageShape_[0],
             Spliterator.ORDERED | Spliterator.IMMUTABLE | Spliterator.NONNULL), true)
-            .forEach(taggedImage ->
-                    StackResampler.this.addImageToRecons((short[]) taggedImage.pix,
-                            (Integer) AcqEngMetadata.getAxes(taggedImage.tags)
-                                     .get(AcqEngMetadata.Z_AXIS)));
+            .forEach(imagePlusFrame ->
+                    StackResampler.this.addImageToRecons(imagePlusFrame.getPixels(),
+                             imagePlusFrame.getFrameNr()));
    }
 
 
+   /**
+    * Helper class to store pixels and frameNr in one instance.
+    */
+   public class ImagePlusFrame {
```

Comment:
"Frame" is often used to mean timepoint in MM so may be good to rename. "Slice"? "Z"?

Reply:
Great catch!  Changed that throughout the code.